go/teams/transactions.go

```diff
@@ -397,28 +397,74 @@ func (tx *AddMemberTx) AddMemberByUsername(ctx context.Context, username string,
 	return err
 }
 
-// AddMemberByAssertion adds an assertion to the team.
+// preprocessAssertion takes an input assertion and determines if this is a valid Keybase-style assertion.
+// If it's an email (or phone) assertion, we assert that it only has one part (and isn't a+b compound).
+// If there is only one factor in the assertion, then that's returned. Otherwise, nil.
+func preprocessAssertion(m libkb.MetaContext, s string) (isEmail bool, single libkb.AssertionURL, err error) {
+	a, err := externals.AssertionParseAndOnly(m.G(), s)
+	if err != nil {
+		return false, nil, err
+	}
+	urls := a.CollectUrls(nil)
+	if len(urls) == 1 {
+		single = urls[0]
+	}
+	for _, u := range urls {
+		if u.IsServerTrust() {
+			isEmail = true
+		}
+	}
+	if isEmail && len(urls) > 1 {
+		return false, nil, NewMixedEmailAssertionError()
+	}
+	return isEmail, single, nil
+}
+
+// AddMemberByAssertionOrEmail adds an assertion to the team. It can handle
+// three major cases:
+//  1. joe OR joe+joe@reddit
+//  2. joe@reddit where
```

Comment:
I don't understand this comment

Reply:
i don't blame you. When writing the comment I went back and changed everything because I understood how broken it was. It's fixed